In [2]:
# importting Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Exploratory Data Analysis

In [11]:
# Load the dataset
df = pd.read_csv("metacritic_pc_games.csv")

df.head()

C:\Users\G1RXK02\AppData\Local\Temp\ipykernel_5952\1507300246.py:2: DtypeWarning: Columns (12) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("metacritic_pc_games.csv")


Game Title                                        Game Poster  \
0  .hackG.U. Last Recode  https://static.metacritic.com/images/products/...   
1  .hackG.U. Last Recode  https://static.metacritic.com/images/products/...   
2  .hackG.U. Last Recode  https://static.metacritic.com/images/products/...   
3  .hackG.U. Last Recode  https://static.metacritic.com/images/products/...   
4  .hackG.U. Last Recode  https://static.metacritic.com/images/products/...   

  Game Release Date  Game Developer                               Genre  \
0          3-Nov-17  CyberConnect2   Genre(s):Miscellaneous,Compilation   
1          3-Nov-17  CyberConnect2   Genre(s):Miscellaneous,Compilation   
2          3-Nov-17  CyberConnect2   Genre(s):Miscellaneous,Compilation   
3          3-Nov-17  CyberConnect2   Genre(s):Miscellaneous,Compilation   
4          3-Nov-17  CyberConnect2   Genre(s):Miscellaneous,Compilation   

                Platforms Product Rating  Overall Metascore  \
0  PlayStation4,Switch,PC              T               69.0   
1  PlayStation4,Switch,PC              T               69.0   
2  PlayStation4,Switch,PC              T               69.0   
3  PlayStation4,Switch,PC              T               69.0   
4  PlayStation4,Switch,PC              T               69.0   

  Overall User Rating  Reviewer Name Reviewer Type  \
0                 8.1           ZTGD        Critic   
1                 8.1        RPGamer        Critic   
2                 8.1   COGconnected        Critic   
3                 8.1  Worth Playing        Critic   
4                 8.1     CGMagazine        Critic   

   Rating Given By The Reviewer Review Date  \
0                          85.0         NaN   
1                          80.0         NaN   
2                          75.0         NaN   
3                          70.0         NaN   
4                          70.0         NaN   

                                         Review Text  
0    The release of .hack//G.U. Last Recode is a ...  
1    .hack//G.U. has remained one of my all-time ...  
2    Last Recode turned out to be something that ...  
3    .hack//G.U. Last Recode is really for the fa...  
4    .hack//G.U. Last Recode is a fascinating nov...

In [12]:
df.sample(5)

Game Title  \
1257                                ADR1FT   
118751                       APB_ Reloaded   
1997    Age of Empires_ Definitive Edition   
362733                           Outriders   
282496            Ghostrunner_ Project_Hel   

                                              Game Poster Game Release Date  \
1257    https://static.metacritic.com/images/products/...         28-Mar-16   
118751  https://static.metacritic.com/images/products/...          1-Dec-11   
1997    https://static.metacritic.com/images/products/...         20-Feb-18   
362733  https://static.metacritic.com/images/products/...          1-Apr-21   
282496  https://static.metacritic.com/images/products/...          3-Mar-22   

                                           Game Developer  \
1257                                      Three One Zero    
118751  Reloaded Productions, Reloaded Games Inc., Rel...   
1997       Microsoft Game Studios, Forgotten Empires LLC    
362733                                    People Can Fly    
282496                                    ONE MORE LEVEL    

                                                    Genre  \
1257     Genre(s):Action,ActionAdventure,General,Survival   
118751  Genre(s):Role-Playing,MassivelyMultiplayerOnli...   
1997                  Genre(s):Strategy,Real-Time,Command   
362733        Genre(s):Action,Shooter,Third-Person,Arcade   
282496                      Genre(s):Action,Platformer,3D   

                                                Platforms Product Rating  \
1257                                      PlayStation4,PC              T   
118751                            PlayStation4,XboxOne,PC              M   
1997                                                   PC              T   
362733  PlayStation4,PlayStation5,Stadia,XboxOne,XboxS...              M   
282496  PlayStation4,PlayStation5,Switch,XboxOne,XboxS...            NaN   

        Overall Metascore Overall User Rating      Reviewer Name  \
1257                 64.0                 5.3     Slant Magazine   
118751               56.0                 5.7           Darkmark   
1997                 69.0                 6.7  The Games Machine   
362733               73.0                 5.7            Xinra68   
282496               76.0                 7.1         Traxillion   

       Reviewer Type  Rating Given By The Reviewer Review Date  \
1257          Critic                          60.0         NaN   
118751          User                           0.0   28-Aug-15   
1997          Critic                          68.0         NaN   
362733          User                           8.0   21-Jun-22   
282496          User                           7.0   12-Mar-22   

                                              Review Text  
1257      Unlike Gravity, which spaced out its most fr...  
118751  A perk that gives immunity to vehicle damage?!...  
1997      After hours spent in the world of Age of Emp...  
362733  I'm playing Outriders on a Steam FREE PLAY eve...  
282496  As good as the original with some of the same ...

## Initial Insights:

- NaN represents missing values
- Nested data in Genre and Platforms
    - separated by ','

In [13]:
df.shape

(513250, 14)

In [14]:
df.columns

Index(['Game Title', 'Game Poster', 'Game Release Date', 'Game Developer',
       'Genre', 'Platforms', 'Product Rating', 'Overall Metascore',
       'Overall User Rating', 'Reviewer Name', 'Reviewer Type',
       'Rating Given By The Reviewer', 'Review Date', 'Review Text'],
      dtype='object')

In [15]:
df.info()       # Data information

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 513250 entries, 0 to 513249
Data columns (total 14 columns):
 #   Column                        Non-Null Count   Dtype  
---  ------                        --------------   -----  
 0   Game Title                    513196 non-null  object 
 1   Game Poster                   513196 non-null  object 
 2   Game Release Date             513196 non-null  object 
 3   Game Developer                512977 non-null  object 
 4   Genre                         513196 non-null  object 
 5   Platforms                     513250 non-null  object 
 6   Product Rating                425527 non-null  object 
 7   Overall Metascore             513181 non-null  float64
 8   Overall User Rating           513128 non-null  object 
 9   Reviewer Name                 512347 non-null  object 
 10  Reviewer Type                 513250 non-null  object 
 11  Rating Given By The Reviewer  510449 non-null  float64
 12  Review Date                   405044 non-nul

## Dimension and Data Type Insights:

- dataset contains - 513250 rows and 14 columns
- 3 numeric columns: Overall Metascore, Overall User Rating, and Rating Given By The Reviewer
    - Overall User Rating will need to be converted
- Two date columns that will need to be converted - Game Release Date and Review Date

In [16]:
df.describe()   # Describe the data

Overall Metascore  Rating Given By The Reviewer
count      513181.000000                 510449.000000
mean           78.038969                     20.304416
std            11.899605                     28.682055
min            16.000000                      0.000000
25%            72.000000                      4.000000
50%            80.000000                      9.000000
75%            86.000000                     10.000000
max            97.000000                    100.000000

In [17]:
df.describe(include = 'all')        # Include the categorical variables

Game Title                                        Game Poster  \
count           513196                                             513196   
unique            5445                                               5437   
top     Cyberpunk 2077  https://static.metacritic.com/images/products/...   
freq             14543                                              14543   
mean               NaN                                                NaN   
std                NaN                                                NaN   
min                NaN                                                NaN   
25%                NaN                                                NaN   
50%                NaN                                                NaN   
75%                NaN                                                NaN   
max                NaN                                                NaN   

       Game Release Date           Game Developer  \
count             513196                   512977   
unique              3155                     2686   
top            10-Dec-20  Blizzard Entertainment    
freq               14640                    38152   
mean                 NaN                      NaN   
std                  NaN                      NaN   
min                  NaN                      NaN   
25%                  NaN                      NaN   
50%                  NaN                      NaN   
75%                  NaN                      NaN   
max                  NaN                      NaN   

                                                    Genre Platforms  \
count                                              513196    513250   
unique                                               1300       231   
top     Genre(s):Action,Shooter,Shooter,First-Person,S...        PC   
freq                                                25649    150738   
mean                                                  NaN       NaN   
std                                                   NaN       NaN   
min                                                   NaN       NaN   
25%                                                   NaN       NaN   
50%                                                   NaN       NaN   
75%                                                   NaN       NaN   
max                                                   NaN       NaN   

       Product Rating  Overall Metascore Overall User Rating Reviewer Name  \
count          425527      513181.000000              513128        512347   
unique              7                NaN                  86        219209   
top                 M                NaN                   7           IGN   
freq           242419                NaN               27764          2942   
mean              NaN          78.038969                 NaN           NaN   
std               NaN          11.899605                 NaN           NaN   
min               NaN          16.000000                 NaN           NaN   
25%               NaN          72.000000                 NaN           NaN   
50%               NaN          80.000000                 NaN           NaN   
75%               NaN          86.000000                 NaN           NaN   
max               NaN          97.000000                 NaN           NaN   

       Reviewer Type  Rating Given By The Reviewer Review Date  \
count         513250                 510449.000000      405044   
unique             2                           NaN        7927   
top             User                           NaN    1-Feb-20   
freq          405329                           NaN        3059   
mean             NaN                     20.304416         NaN   
std              NaN                     28.682055         NaN   
min              NaN                      0.000000         NaN   
25%              NaN                      4.000000         NaN   
50%              NaN                      9.000000         NaN   


## Descriptive Statistics Insights:

- mean overall metascore is close to 50th percentile
- big difference between mean and max for Rating by the Reviewer - possible outliers
- 1 column has 2 unique values - Reviewer Type

In [18]:
df['Reviewer Type'].nunique()    # Number of unique values

2

In [19]:
df['Reviewer Type'].unique()    # Unique values

array(['Critic', 'User'], dtype=object)

In [20]:
df['Reviewer Type'].value_counts(dropna = False)

Reviewer Type
User      405329
Critic    107921
Name: count, dtype: int64

The two reviewer types are users and critics. The majority of the reviewer types are users. 

Investigating Null values and deciding what to do with them.

In [21]:
df['Review Date'].nunique()     # Number of unique values

7927

In [22]:
df['Review Date'].isna().sum()      # Count how many null values there are

108206

In [26]:
df['Review Date'].value_counts(dropna = False, ascending = False)

Review Date
NaN          108206
1-Feb-20       3059
12-Dec-20      2914
2-Feb-20       2673
31-Jan-20      2102
              ...  
28-Dec-03         1
3-Mar-04          1
14-Dec-02         1
16-Jul-02         1
19-Jul-03         1
Name: count, Length: 7928, dtype: int64

In [27]:
df.isna().sum()     # Count null values for all columns

Game Title                          54
Game Poster                         54
Game Release Date                   54
Game Developer                     273
Genre                               54
Platforms                            0
Product Rating                   87723
Overall Metascore                   69
Overall User Rating                122
Reviewer Name                      903
Reviewer Type                        0
Rating Given By The Reviewer      2801
Review Date                     108206
Review Text                        446
dtype: int64

In [28]:
df.isna().sum().sum()     # Count null values for whole dataset

200759

In [31]:
percent_not_null_review_date = round((len(df) - df['Review Date'].isna().sum()) / len(df), 3) * 100

In [35]:
print("% of review date not null: ", percent_not_null_review_date, "%")

% of review date not null:  78.9 %
